sdk/cargo-build-sbf/src/main.rs

```diff
@@ -524,8 +525,19 @@ fn build_sbf_package(config: &Config, target_directory: &Path, package: &cargo_m
         env::set_var("RUSTFLAGS", &rustflags);
     }
     if config.verbose {
-        debug!("RUSTFLAGS={}", &rustflags);
-    };
+        debug!(
+            "RUSTFLAGS=\"{}\"",
+            env::var("RUSTFLAGS").ok().unwrap_or_default()
+        );
+    }
+
+    // RUSTC variable overrides cargo +<toolchain> mechanism of
+    // selecting the rust compiler and makes cargo run a rust compiler
+    // other than the one linked in BPF toolchain. We have to prevent
+    // this by removing RUSTC from the child process environment.
+    if env::var("RUSTC").is_ok() {
+        env::remove_var("RUSTC")
```

Comment:
should probably log a warning about ignoring the caller's environment?

Reply:
good poinr, i’ll add a warning message.